Doc/library/asyncio-pools.rst

```diff
@@ -0,0 +1,95 @@
+.. currentmodule:: asyncio
+
+.. versionadded:: 3.9
+
+.. _asyncio-pools:
+
+=====
+Pools
+=====
+
+**Source code:** :source:`Lib/asyncio/pools.py`
+
+-----------------------------------------------
+
+.. note::
+    This section of the documentation and all of its members have been
+    added *provisionally* to asyncio's API. For more details, see
+    :term:`provisional api`.
+
+Asyncio pools are high-level, asynchronous context managers that can be used
+to concurrently run blocking functions and methods.
+
+There are many potential use cases, but a particularly useful one is for
+combining libraries without asyncio support with existing asyncio programs.
+Normally, calling a non-async function within the event loop would will result
+in blocking the event loop until the function returns. However, by using a
+pool to run the function, it can be executed in a separate worker (such as a
+thread or process) without blocking the event loop.
+
+.. class:: ThreadPool(concurrency=None)
+
+   An asynchronous thread pool that provides methods to concurrently
+   run IO-bound functions, without blocking the event loop.
+
+   *concurrency* is an optional argument that limits the number of
+   threads to utilize in the thread pool. With the default value of
+   ``None``, the amount of threads used will scale based on the
+   number of processors.
+
+   .. coroutinemethod:: run(/, func, *args, **kwargs)
+
+      Asynchronously run *func* with its arguments and keyword-arguments
+      within the thread pool, and return a :class:`asyncio.Future` object
+      that represents the eventual result of its execution. ::
+
+          async with asyncio.ThreadPool() as pool:
+              await pool.run(time.sleep, 1)
+
+      Raises a :exc:`RuntimeError` if the thread pool is *not* running.
+
+   .. coroutinemethod:: astart()
+
+      Schedule the start of the thread pool and spawn its threads. Note that
```

Comment:
```suggestion
      Start the thread pool and spawn its threads. Note that
```

With how coroutines are typically (with an `await` expression) it's not about "scheduling an action" it's about "doing an action". I admit the details can be blurry for some people, but to make it simpler I suggest to simply describe what coroutines are doing without mentioning "scheduling"

Reply:
Okay, that seems reasonable to me. The use of "scheduling" was mainly there as a way to describe it as "begin this operation, but without blocking". 

However, that might be needlessly over-complicating the documentation. The fact that it's a coroutine method instead of a subroutine method should hopefully explain that to some degree. So, I'll change it as suggested.